site/src/xServices/workspace/workspaceXService.ts

```diff
@@ -196,38 +198,33 @@ export const workspaceMachine = createMachine(
           ],
         },
       },
+      sseFailure: {
+        entry: ["assignRefreshWorkspaceError", "assignGetResourcesError"],
+        after: {
+          "1000": {
+            target: 'ready.listenForEvents'
+          }
+        }
+      },
       ready: {
         type: "parallel",
         states: {
-          pollingWorkspace: {
-            initial: "refreshingWorkspace",
-            states: {
-              refreshingWorkspace: {
-                entry: "clearRefreshWorkspaceError",
-                invoke: {
-                  src: "refreshWorkspace",
-                  id: "refreshWorkspace",
-                  onDone: [
-                    {
-                      actions: ["refreshTimeline", "assignWorkspace"],
-                      target: "waiting",
-                    },
-                  ],
-                  onError: [
-                    {
-                      actions: "assignRefreshWorkspaceError",
-                      target: "waiting",
-                    },
-                  ],
-                },
+          listenForEvents: {
+            on: {
+              UPDATE_EVENT: {
+                actions: ["updateWorkspace", "assignResources"]
               },
-              waiting: {
-                after: {
-                  "1000": {
-                    target: "refreshingWorkspace",
-                  },
-                },
+              SSE_ERROR: {
+                target: "#workspaceState.sseFailure"
               },
+              CHECK_REFRESH_TIMELINE: {
+                actions: ["refreshTimeline"]
+              }
+            },
+            entry: ["clearGetWorkspaceError", "clearGetResourcesError", "initSse"],
```

Comment:
whichever way you decide to go with the errors above, make sure it matches what you're clearing here